src/backends/policy_engine/souffle/raksha_datalog_facts_test.cc

```diff
@@ -52,6 +54,18 @@ class DumpFactsToDirectoryTest : public testing::Test {
       is_operation_file_facts_.push_back(fact.ToDatalogFactsFileString());
       facts_.AddIsOperationFact(std::move(fact));
     }
+    directory_ = common::utils::CreateTemporaryDirectory();
+  }
+
+  // Usually, the test fixture's destructor is fine. But I'm pretty sure that
+  // `remove_all` can throw an exception in general, which would cause undefined
+  // behavior if we put this in the destructor. This probably isn't possible
+  // with the noexception flags that we're using, but when it comes to undefined
+  // behavior in dtors, better safe than sorry.
+  void TearDown() override {
+    if (!directory_.empty()) {
+      std::filesystem::remove_all(directory_);
+    }
```

Comment:
If it is temporary directory, do we really need to clean it up explicitly?

Reply:
Yes, because it isn't temporary in the sense of "will be cleaned up when all references to it expire", it's more likely temporary in the sense of "will be cleaned up when the machine is powered off". Garbage can accumulate in such directories surprisingly quickly, especially for devices that people don't think to turn off frequently (like phones).